packages/core/src/types.ts

```diff
@@ -2173,25 +2173,11 @@ export type SnapshotFrom<T> = ReturnTypeOrValue<T> extends infer R
     ? TSnapshot
     : R extends Actor<infer TLogic>
       ? SnapshotFrom<TLogic>
-      : R extends StateMachine<
```

Comment:
There was a reason why this had two separate branches for `StateMachine` and `ActorLogic`, although I don't recall what that was right now. I'd probably prefer to leave this change for another PR